File: codigo-antigo-curso/3-semestre/cliente-servidor/InvocationResult.py
listaRestaurantes = {
    "donnaPizza": 500,
    "cleitonSalgados": 350,
    "padariaJose": 750,
    }
clienteEstaEm = 1000;
menorDistancia = float('inf');
restauranteMaisProximo = "";

for nome, distancia in listaRestaurantes.items():
    resultado = clienteEstaEm - distancia;
    if resultado < menorDistancia:
        menorDistancia = resultado;
        restauranteMaisProximo = nome;

print("Restaurante mais próximo:", restauranteMaisProximo, "-", menorDistancia, "metros");

File: codigo-antigo-curso/3-semestre/cliente-servidor/InvocationResult.java
import java.util.HashMap;
import java.util.Map;

public class InvocationResult {
    public static void main (String[] args){
        HashMap<String, Integer> listaRestaurantes = new HashMap<>();
        listaRestaurantes.put("donnaPizza", 500);
        listaRestaurantes.put("cleitonSalgados", 350);
        listaRestaurantes.put("padariaJose", 750);
        int clienteEstaEm = 1000;
        double menorDistancia = Double.MAX_VALUE;
        String restauranteMaisProximo = "";

        for (Map.Entry<String, Integer> entry : listaRestaurantes.entrySet()) {
            String nome = entry.getKey();
            int distancia = entry.getValue();
            int resultado = clienteEstaEm - distancia;
            if (resultado < menorDistancia) {
            menorDistancia = resultado;
            restauranteMaisProximo = nome;}
        }
        System.out.println("O restaurante mais proximo é o " + restauranteMaisProximo + " e ele está a " + (int)menorDistancia + " Metros");
    }
}
